tests/Functional/Admin/BlockAdminTest.php

```diff
@@ -249,15 +249,16 @@ private function prepareData(): void
 
         $parentBlock = new SonataPageBlock();
         $parentBlock->setType('sonata.page.block.container');
-        $parentBlock->setPage($page);
 
         $parentBlock2 = new SonataPageBlock();
         $parentBlock2->setType('sonata.page.block.container');
-        $parentBlock2->setPage($page);
 
         $block = new SonataPageBlock();
         $block->setType('sonata.block.service.text');
-        $block->setParent($parentBlock);
+
+        $page->addBlock($parentBlock);
+        $page->addBlock($parentBlock2);
+        $page->addBlock($block);
```

Comment:
Is better to add blocks like this, otherwise they don't get related on the $page side, I think it is because how doctrine works, maybe the object is not refreshed, anyway, better do it this way.

Reply:
When you add, it also call the other setter. But when you set, it doesn't.